### PR TITLE
Try to clean this up

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Role Name
```
Comment (ghost) [inline README.md:1]: Write this - maybe?

File: meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+galaxy_info:
```
Comment (ghost) [inline meta/main.yml:1]: Nope

File: meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  license: CC-BY-SA
+
+  min_ansible_version: 2.2
+
+  platforms:
+    - name: Ubuntu
+      versions:
+        - yakkety
+    - name: Debian
```
Comment (ghost) [inline meta/main.yml:14]: Nope

File: meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      versions:
+        - jessie
+
+  galaxy_tags:
+    - stuvus
```
Comment (ghost) [inline meta/main.yml:19]: Nope

File: tasks/install.yml [hunk 5]
```diff
@@ -0,0 +1,151 @@
+---
+- name: download file from a file path
```
Comment (ghost) [inline tasks/install.yml:2]: Capitalize

File: tasks/install.yml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  get_url:
+    url: https://download.nextcloud.com/server/releases/nextcloud-11.0.3.tar.bz2
```
Comment (ghost) [inline tasks/install.yml:4]: Softcode & Move to the control machine - we might not have internet on the server

File: tasks/install.yml [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    dest: /tmp/nextcloud.tar.bz2
+
+- name: Create root
+  file: 
+    path: "{{nextcloud_path}}"
```
Comment (ghost) [inline tasks/install.yml:9]: Add spaces between word and the curly braces

File: tasks/install.yml [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    state: directory 
+    mode: 0755
+    owner: www-data
+
+- unarchive:
+    src: /tmp/nextcloud.tar.bz2
+    dest: /var/www
+    remote_src: True
+    owner: www-data
```
Comment (ghost) [inline tasks/install.yml:18]: Softcode this

File: tasks/install.yml [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    group: www-data
+
+- name: Create a new database with name '{{ nextcloud_mysql_database_name }}'
+  mysql_db:
```
Comment (ghost) [inline tasks/install.yml:22]: PostgreSQL and softcode

File: tasks/install.yml [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+- name: Create mariadb user
+  mysql_user:
+    name: "{{ nextcloud_mysql_user }}"
+    password: "{{ nextcloud_mysql_password }}"
+    priv: '{{ nextcloud_mysql_database_name }}.*:ALL,GRANT'
+    state: present
+
+- name: Run the nextcloud installer from command line using sudo since become is shitty
+  shell: sudo -u www-data php occ  maintenance:install --database "mysql" --database-name "{{nextcloud_mysql_database_name}}"  --database-user "{{nextcloud_mysql_user}}" --database-pass "{{nextcloud_mysql_password}}" --admin-user "{{ nextcloud_user }}" --admin-pass "{{ nextcloud_password }}"
```
Comment (ghost) [inline tasks/install.yml:34]: Use the become module and softcode this

File: tasks/install.yml (hunk 9 above)
Comment (ghost) [reply tasks/install.yml:22]: Also, this isn't what the role is supposed to do

File: tasks/install.yml [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+  args:
+    chdir: "{{ nextcloud_path }}"
+
+- name: Add the domain to the trusted domains
+  shell: sudo -u www-data php occ config:system:set trusted_domains 2 --value=nextcloud.stuvus.de
```
Comment (ghost) [inline tasks/install.yml:39]: WHAT THE FUCK
DON'T HARDCODE STUVUS URLS HERE

File: tasks/install.yml [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+  args:
+    chdir: "{{ nextcloud_path }}"
+
+- name: Enable the ldap plugin from command line using sudo since become is shitty
```
Comment (ghost) [inline tasks/install.yml:43]: Remove the useless comment in this name

File: tasks/install.yml [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+    chdir: "{{ nextcloud_path }}"
+
+- name: Create empty ldap config
+  shell: sudo -u www-data php occ ldap:create-empty-config
+  args:
+    chdir: "{{ nextcloud_path }}"
+  register: config_id_output
+
+- set_fact:
+    config_id: "{{ config_id_output['stdout'].split(' ') | last}}"
```
Comment (ghost) [inline tasks/install.yml:55]: Space at the end

File: tasks/install.yml [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+  shell: "sudo -u www-data php occ ldap:set-config {{ config_id }} ldapPort {{ ldapPort }}"
+  args:
+    chdir: "{{ nextcloud_path }}"
+
+- name: Set ldapEmailAttribute
+  shell: "sudo -u www-data php occ ldap:set-config {{ config_id }} ldapEmailAttribute {{ ldapEmailAttribute }}"
+  args:
+    chdir: "{{ nextcloud_path }}"
+
+
```
Comment (ghost) [inline tasks/install.yml:142]: Remove this empty line

File: tasks/install.yml [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+  shell: "sudo -u www-data php occ ldap:set-config {{ config_id }} ldapHost {{ ldapHost }}"
+  args:
+    chdir: "{{ nextcloud_path }}"
+
+- name: Set ldapLoginFilter
+  shell: "sudo -u www-data php occ ldap:set-config {{ config_id }} ldapLoginFilter '{{ ldapLoginFilter }}'"
+  args:
+    chdir: "{{ nextcloud_path }}"
+
+- name: Set ldapUserDisplayName
```
Comment (ghost) [inline tasks/install.yml:112]: Why don't you just iterate over all variables?

File: tasks/main.yml [hunk 16]
```diff
@@ -0,0 +1,23 @@
+---
+- name: Update repositories cache and install needed php packages
+  apt:
+    name: "{{item}}"
```
Comment (ghost) [inline tasks/main.yml:4]: Spaces

File: tasks/main.yml [hunk 17]
```diff
@@ -0,0 +1,23 @@
+---
+- name: Update repositories cache and install needed php packages
```
Comment (ghost) [inline tasks/main.yml:2]: php -> PHP

File: tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+    - php-zip
+    - php-curl
+    - php-mbstring
+    - php-ldap
+    - python-mysqldb
+    - php-mysql
+    - php-xml
+
+- name: Run only if it hasn't been done already
+  stat: path={{nextcloud_path}}
```
Comment (ghost) [inline tasks/main.yml:18]: Don't use this syntax

**Review by ghost — COMMENTED** [review]
Fix this ASAP

File: meta/main.yml (hunk 2 above)
Comment (SlothOfAnarchy) [reply meta/main.yml:1]: Copy the file from ansible_skeleton and populate it.

File: tasks/main.yml [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+  register: st
+
+- include: install.yml
```
Comment (SlothOfAnarchy) [inline tasks/main.yml:21]: deprecated

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

**Comment by dasJ** [comment]
Automatically bumping this issue because it was inactive for 26 days.

**Comment by SlothOfAnarchy** [comment]
@Scriptkiddi bump

**Comment by dasJ** [comment]
Automatically bumping this issue because it was inactive for 48 days.

**Comment by SlothOfAnarchy** [comment]
@Scriptkiddi what's the state on this?

**Comment by dasJ** [comment]
Automatically bumping this issue because it was inactive for 15 days.

**Pull request closed without merging**